builders/tools/explorers/verify-contract.md

```diff
@@ -0,0 +1,102 @@
+    ![First Page Screenshot](/images/builders/tools/explorers/verify-contract/verify-contract-2.png)
+7. On the second page, the **Contract Address**, **Compiler**, and **Constructor Arguments** fields should be prefilled. Copy and paste the entirely of the contract's content into the text field labeled as such
+8. (Optional) Select **Yes** for **Optimization** if it was enabled during compilation, and filled in the number of runs under **Misc Settings/Runs(Optimizer)**
+9. (Optional) Add contract libraries and their addresses if any were used in the contract
+10. Check any other optional fields that may apply to your contract, and fill them out accordingly
+11. Click on the CAPTCHA at the bottom and the **Verify and Publish** button to confirm and begin verification
+    ![Second Page Screenshot](/images/builders/tools/explorers/verify-contract/verify-contract-3.png)
+12. After a short wait, the result of verification will be displayed in the browser, and a success result page will display the contract's ABI encoded constructor arguments, the contract name, bytecode, and ABI
+    ![Result Page Screenshot](/images/builders/tools/explorers/verify-contract/verify-contract-4.png)
+
+### Verify on BlockScout {: #verify-on-blockscout }
+
+1. Go to the contract's page on [BlockScout](https://moonbase-blockscout.testnet.moonbeam.network/) for the respective network by searching for its address, and click on **Verify & Publish** under the **Code** tab
+    ![BlockScout Verify Button](/images/builders/tools/explorers/verify-contract/verify-contract-5.png)
+2. On the verification page, the contract address will be prefilled. Fill in the contract name; this must be the same name as the contract definition. In the example, the contract name is `Incrementer`
+3. Fill in the **Compiler**, **EVM Version**, and **Optimization** fields (if optimization was enabled during compiling)
+4. Copy and paste the entirety of the Solidity smart contract into the text field
+5. (Optional) Toggle **Try to fetch constructor arguments automatically** to **Yes**, or fill in the ABI-encoded constructor arguments manually if the contract constructor accepts arguments
+5. (Optional) Add contract libraries and their addresses if any were used in the contract
+6. Click **Verify & Publish** at the bottom when all the information is filled out
+    ![BlockScout Verify Page](/images/builders/tools/explorers/verify-contract/verify-contract-6.png)
+7. If the verification is completely successfully, the browser will return to the contract's **Code** page, displaying information including the contract's ABI encoded constructor arguments, the contract name, bytecode, ABI and source code. The contract page will also have two additional tabs, **Read Contract** and **Write Contract** for users to read or write to the contract directly
+    ![BlockScout Result Page](/images/builders/tools/explorers/verify-contract/verify-contract-7.png)
```

Comment:
More zoom in this image. You can merge multiple images so they are bigger vertically. Ping me for help